Lib/test/test_re.py

```diff
@@ -883,31 +883,137 @@ def test_named_unicode_escapes(self):
         self.checkPatternError(br'\N{LESS-THAN SIGN}', r'bad escape \N', 0)
         self.checkPatternError(br'[\N{LESS-THAN SIGN}]', r'bad escape \N', 1)
 
-    def test_string_boundaries(self):
+    def test_word_boundaries(self):
```

Comment:
[no change needed] I think we _could_ use subtests to provide more informative context (basically what you've got in each # comment header) when a test fails? but this is already in keeping with the existing style of this file, so not a big deal.  these seem to nicely encode the current behavior state of our `re` implementation.

Reply:
The comments provide context to the reader of the code (so you will know where to add new tests), not when a test fails. The traceback already has all necessary context.

`subTest()` has two functions:
* Provide more informative context when a test fails. This is especially useful when traceback does not identify the failed test (in a loop, with generated test data).
* Allows to continue execution when a test fails and collect information about other test failures in one run. This works only when tests are independent and `subTest()` is well granulated.

It has a drawback -- the traceback is limited and can be less informative if `subTest()` is used in a helper.